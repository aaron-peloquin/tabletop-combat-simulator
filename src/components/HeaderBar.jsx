
export default () => <p>Header Bar</p>