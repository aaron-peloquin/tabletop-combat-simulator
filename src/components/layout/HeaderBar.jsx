import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import { connect } from "react-redux";

import ToggleSidebar from "./../../store/dispatcher/ToggleSidebar";

const HeaderBar = (props) => {
  const {dispatch} = props;
  return <AppBar>
    <Toolbar>
      <IconButton color="inherit" aria-label="Menu" onClick={()=>{ return ToggleSidebar(dispatch); }} >
        <MenuIcon />
      </IconButton>
      <Typography variant="h6" color="inherit">Tabletop Combat Simulator</Typography>
    </Toolbar>
  </AppBar>;
};

const mapStateToProps = (state) => {
  return {open:state.sideBar};
};

export default connect(mapStateToProps)(HeaderBar);