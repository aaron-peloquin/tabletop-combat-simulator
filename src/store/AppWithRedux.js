import React from 'react'
import {initializeStore} from './store'

const isServer = typeof window === 'undefined'
const reduxStoreSymbol = Symbol("reduxStore")

function getOrCreateStore (initialState) {
  // Always make a new store if server, otherwise state is shared between requests
  if (isServer) {
    return initializeStore(initialState)
  }

  // Create store if unavailable on the client and set it on the window object
  if (!window[reduxStoreSymbol]) {
    window[reduxStoreSymbol] = initializeStore(initialState)
  }
  return window[reduxStoreSymbol]
}

export default (App) => {
  return class AppWithRedux extends React.Component {
    static async getInitialProps (appContext) {
      // Get or Create the store with `undefined` as initialState
      // This allows you to set a custom default initialState
      const reduxStore = getOrCreateStore()

      // Provide the store to getInitialProps of pages
      appContext.ctx.reduxStore = reduxStore

      let appProps = {}
      if (typeof App.getInitialProps === 'function') {
        appProps = await App.getInitialProps(appContext)
      }

      return {
        ...appProps,
        initialReduxState: reduxStore.getState()
      }
    }

    constructor (props) {
      super(props)
      this.reduxStore = getOrCreateStore(props.initialReduxState)
    }

    render () {
      return <App {...this.props} reduxStore={this.reduxStore} />
    }
  }
}
