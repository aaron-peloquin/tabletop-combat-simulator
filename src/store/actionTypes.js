/**
 * This is an object of Redux Actions. It mostly exists to
 *  ensure we do not create duplicate actions, but it can
 *  also be used as a list of our existing actions
*/
const actions = {
  CopyCreature: 'CopyCreature',
  DeleteAllCreatures: 'DeleteAllCreatures',
  DeleteCreature: 'DeleteCreature',
  SaveCreature: 'SaveCreature',
  SetEditCreature: 'SetEditCreature',
  ToggleSidebar: 'ToggleSidebar',
  RunSimulation: 'RunSimulation',
  UpdateEditCreature: 'UpdateEditCreature',
}

export default actions
