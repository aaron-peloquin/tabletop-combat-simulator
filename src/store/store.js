import {createStore, applyMiddleware, combineReducers} from "redux"
import {composeWithDevTools} from "redux-devtools-extension"
import thunkMiddleware from "redux-thunk"

/** Reducers */
import combat from "./reducer/combat"
import creatures from "./reducer/creatures"
import sideBar from "./reducer/sideBar"
import editing from "./reducer/editing"

/** Reducers */
const reducers = combineReducers({
  combat,
  creatures,
  editing,
  sideBar,
})

/**
 * Initializes the store
 * @param {obj} initialState the state our app starts with
 * @return {obj} the redux store
 */
export function initializeStore(initialState = initialState) {
  return createStore(reducers, initialState, composeWithDevTools(applyMiddleware(thunkMiddleware)))
}
