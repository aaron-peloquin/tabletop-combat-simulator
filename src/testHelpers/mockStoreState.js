export default {
  "creatures": [
    {
      "name": "Goplin",
      "cr": "1/4",
      "hash": "hivn3p0k",
      "description": "A stout monster",
      "defaultInitiative": 10,
      "armor": 11,
      "hp": 5,
      "hitDiceEquation": "1d20+1",
      "damageDiceEquation": "1d4"
    },
    {
      "name": "Elven Ranger",
      "cr": "1/2",
      "hash": "h3dxf5wl",
      "description": "A swift hunter",
      "defaultInitiative": 16,
      "armor": 13,
      "hp": 10,
      "hitDiceEquation": "1d20+8",
      "damageDiceEquation": "1d6+3"
    },
    {
      "name": "Knight",
      "cr": "1",
      "hash": "hee9p286",
      "description": "An honorable soldier",
      "defaultInitiative": 11,
      "armor": 18,
      "hp": 25,
      "hitDiceEquation": "1d20+4",
      "damageDiceEquation": "1d12+4"
    },
    {
      "name": "Lizardfolk",
      "cr": "3",
      "hash": "howx26rt",
      "description": "An ancient sneaky monster",
      "defaultInitiative": 14,
      "armor": 14,
      "hp": 15,
      "hitDiceEquation": "1d20+5",
      "damageDiceEquation": "1d10+2"
    }
  ],
  "sideBar": false,
}