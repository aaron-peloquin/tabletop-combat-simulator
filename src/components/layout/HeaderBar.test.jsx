import { Provider } from "react-redux"
import { createShallow } from "@material-ui/core/test-utils"
import toJson from "enzyme-to-json"
import { initializeStore } from "../../store/store"

import HeaderBar, { mapStateToProps } from "./HeaderBar"
import mockStoreState from "./../../testHelpers/mockStoreState"

describe("<HeaderBar />", ()=>{
  const RenderShallowUntilComponent = createShallow({"untilSelector":"HeaderBar"})
  let store, props, Component
  beforeEach(async ()=>{
    store = initializeStore(mockStoreState)
    props = mapStateToProps(store.getState())
    /** Attach spy to store.dispatch */
    store.dispatch = jest.fn(store.dispatch)
    Component = RenderShallowUntilComponent(<Provider store={store}><HeaderBar {...props} /></Provider>)
  })

  it("loads", async () => {
    expect(typeof Component).toBe("object")
  })

  it("has an IconButton with prop of aria-label=Menu", () => {
    expect(Component.find("WithStyles(IconButton)[aria-label=\"Menu\"]").length).toBe(1)
  })
  it("has a state of open=false by default", () => {
    expect(store.getState().sideBar).toBe(false)
  })

  it("calls function, dispatch, and updates state on click of menu", () => {
    Component.find("WithStyles(IconButton)[aria-label=\"Menu\"]").simulate("click")
    expect(store.dispatch).toBeCalledTimes(1)
    expect(store.getState().sideBar).toBe(true)
  })

  it("snapshots", () => {
    expect(toJson(Component)).toMatchSnapshot()
  })
})