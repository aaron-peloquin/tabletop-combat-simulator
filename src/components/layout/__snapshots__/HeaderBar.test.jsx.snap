// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderBar /> snapshots 1`] = `
<WithStyles(AppBar)>
  <WithStyles(Toolbar)>
    <WithStyles(IconButton)
      aria-label="Menu"
      color="inherit"
      onClick={[Function]}
    >
      <pure(MenuIcon) />
    </WithStyles(IconButton)>
    <WithStyles(Typography)
      color="inherit"
      variant="h4"
    >
      Tabletop Combat Simulator
    </WithStyles(Typography)>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
